### DISCUSSION TITLE
Adding New Objects

### DISCUSSION THREAD

**Discussion by gdismond** [allenai/ai2thor#622] [Q&A]
I noticed that someone had a question about adding new objects back in June of 2018 https://github.com/allenai/ai2thor/issues/60. Does a document exist for hooking up a new object after it has been created? (i.e. establishing all of the script and data dependencies required to make the object function according to spec in the environment)

If no documentation exists, could someone please give contact information for a person that could help with this goal?

Thank you!

**Comment by winthos** [comment]
No such documentation currently exists for a way to set up an entirely new object.

Depending on the object type that you are trying to add, the process can be relatively straightforward to extremely involved. Objects that have things like moving parts, or have multiple interaction properties are quite extensive in their setup. I can help to answer any specific questions you might add, and potentially walk you through the process.

Note that being familiar with the `Unity` engine will be a must in order to set up new objects completely from scratch. This goes beyond assigning the correct component scripts to objects, as involved use of the Heirarchy tab, as well as the Unity Prefab pipeline will be required. Additionally components will likely need to be adjusted or assigned to manually in cases like making a new microwave with a door that can open.
